packages/package_qrcode_5_1/tool_dependencies.xml

```diff
@@ -0,0 +1,17 @@
+<tool_dependency>
+    <package name="qrcode" version="5.1">
+        <install version="1.0">
+            <actions>
+                <action type="download_by_url">https://codeload.github.com/lincolnloop/python-qrcode/tar.gz/master</action>
```

Comment:
Same as with image, better to use

```xml
         <action type="setup_python_environment">
               <repository name="package_python_2_7" owner="iuc">
                   <package name="python" version="2.7" />
               </repository>
               <package">https://codeload.github.com/lincolnloop/python-qrcode/tar.gz/master</package>
         </action>
```

Same note applies, the link should normally guarantee you to get version 5.1 of qrcode.
Perhaps clone with git and fix at a specific revision?